web/user.go

```diff
@@ -133,6 +124,19 @@ func loginWithTumCredentials(username, password string, usersDao dao.UsersDao) (
 }
 
 func (r mainRoutes) LoginPage(c *gin.Context) {
+	redirUrlStr := "/"
+	redirURL, err := getRedirectUrl(c)
```

Comment:
I think this will always be `/` since 
```golang
ret := c.Request.FormValue("return") == ""
ref := c.Request.FormValue("ref") == ""
```
since these are only sent on form.submit:
```html
<form x-data="{ref: global.getLoginReferrer()}" id="loginForm" method="post" class="grid gap-3 py-4 px-5">
                <input x-model="ref" type="hidden" name="ref">
                ....
</form>
```

Maybe I'm also wrong? 🤔

Reply:
It worked when I tested it, but I think we're overcomplicating it here frontend wise. I'll rework it a little.

Reply:
Does the TUM shibboleth allow a [relay state](https://stackoverflow.com/questions/34350160/what-is-exactly-relaystate-parameter-used-in-sso-ex-saml)? Because if so, we might want to make use of this feature.

Reply:
That's a great impulse, will check it out!

Reply:
Our SAML library already uses the relay state to track requests, hacking in the referrer won't work without significant changes. Also, the relay state can only store up to 80 bytes. Therefore, I decided to stick with our cookie workaround which seems to be an established way of doing this.